hmrc-mongo-play-27/src/main/scala/uk/gov/hmrc/mongo/play/json/formats/MongoUuidFormats.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.mongo.play.json.formats
+
+import org.bson.BsonBinarySubType
+import org.bson.UuidRepresentation
+import org.bson.internal.UuidHelper
+import play.api.libs.json._
+
+import java.util.Base64
+import java.util.UUID
+
+abstract class MongoUuids(uuidRep: UuidRepresentation) {
+  outer =>
+
+  private val encoder = Base64.getEncoder
+  private val decoder = Base64.getDecoder
+
+  private val UuidOldSubtype = BsonBinarySubType.UUID_LEGACY.getValue
+  private val UuidNewSubtype = BsonBinarySubType.UUID_STANDARD.getValue
+
+  private val ChosenSubType =
+    if (uuidRep == UuidRepresentation.JAVA_LEGACY) UuidOldSubtype else UuidNewSubtype
```

Comment:
`STANDARD` is subtype 4, 
`C_SHARP_LEGACY`, `JAVA_LEGACY` & `PYTHON_LEGACY` are subtype 3. 

There is `UNSPECIFIED` too. 

```
if (uuidRep == STANDARD) ... 
```
This would be a bit more robust. I wonder whether we should handle `UNSPECIFIED` by throwing an exception?

Reply:
Do we ever expect to encounter data written by the C# or Java drivers in HMRC databases?

Reply:
I have changed this logic around because what you suggest definitely makes more sense, but I have not done anything to address the `UNSPECIFIED` encoding because this value actually comes from the abstract class parameter, so I think it would take ermm...let's say an unusual level of self-sabotage for users to cause themselves a problem with this one 😆 They would have to declare their own subtype of `MongoUuids` that uses a different representation.
Hope that's OK!